docs/sphinx/using/applications.rst

```diff
@@ -1,31 +1,205 @@
-    /applications/python/shors.ipynb
-    /applications/python/hybrid_qnns.ipynb
-    /applications/python/qaoa.ipynb
-    /applications/python/digitized_counterdiabatic_qaoa.ipynb
-    /applications/python/krylov.ipynb
-    /applications/python/afqmc.ipynb
-    /applications/python/quantum_fourier_transform.ipynb
-    /applications/python/quantum_teleportation.ipynb
-    /applications/python/quantum_volume.ipynb
-    /applications/python/readout_error_mitigation.ipynb
-    /applications/python/vqe.ipynb
-    /applications/python/vqe_advanced.ipynb
-    /applications/python/hadamard_test.ipynb
-    /applications/python/logical_aim_sqale.ipynb
+====================
+
+.. toctree::
+   :maxdepth: 1
+   :hidden:
+
+   /applications/python/qaoa
+   /applications/python/digitized_counterdiabatic_qaoa
+   /applications/python/krylov
+   /applications/python/bernstein_vazirani
+   /applications/python/cost_minimization
+   /applications/python/deutsch_jozsa
+   /applications/python/divisive_clustering_coresets
+   /applications/python/hadamard_test
+   /applications/python/logical_aim_sqale
+   /applications/python/hamiltonian_simulation
+   /applications/python/quantum_fourier_transform
+   /applications/python/quantum_teleportation
+   /applications/python/quantum_volume
+   /applications/python/readout_error_mitigation
+   /applications/python/unitary_compilation_diffusion_models
+   /applications/python/vqe_advanced
+   /applications/python/afqmc
+
+
+
+.. raw:: html
+
+    <div class="filter-groups" style="margin-bottom: 60px;">
+        <div class="tag-filters">
+            <h3>Filter by Domain:</h3>
+            <button class="tag-button active" data-group="domain" data-tag="all">All</button>
+            <button class="tag-button" data-group="domain" data-tag="optimization">Optimization</button>
+            <button class="tag-button" data-group="domain" data-tag="chemistry">Chemistry</button>
+            <button class="tag-button" data-group="domain" data-tag="fundamental">Fundamental Algorithms</button>
+            <button class="tag-button" data-group="domain" data-tag="aiforq">AI Supercomputing for Quantum</button>
+            <button class="tag-button" data-group="domain" data-tag="qforai">Quantum for AI</button>
+        </div>
+        
+        <div class="tag-filters" style="margin-top: 20px;">
+            <h3>Specialized Backends:</h3>
+            <button class="tag-button active" data-group="backend" data-tag="all">All</button>
+            <button class="tag-button" data-group="backend" data-tag="mqpu">MQPU</button>
+            <button class="tag-button" data-group="backend" data-tag="mgpu">MGPU</button>
+            <button class="tag-button" data-group="backend" data-tag="noise">Noisy</button>
+            <button class="tag-button" data-group="backend" data-tag="qpu">QPU</button>
+        </div>
+
+        <div class="tag-filters" style="margin-top: 20px;">
+            <h3>CUDA-X Library:</h3>
```

Comment:
I'm not sure if we need that, in the same way that we won't add a sample/observe tags, it's an implementation detail, the assumption is that the applications are written in the best way to utilize libraries and CUDA-Q APIs. I would add a QEC tag to the domain filters